code-challenges.test.js

```diff
@@ -14,92 +14,246 @@
 
-var testString1 = "albatross"
-// Expected output: "aLbAtRoSs"
-var testString2 = "jabberwocky"
-// Expected output: "jAbBeRwOcKy"
 
 
 
 
 
+// // --------------------4) Create a function that takes in a string and returns a string with every other letter capitalized.
+
+// // a) Create a test with expect statements for each of the variables provided.
+
+// var testString1 = "albatross"
+// // Expected output: "aLbAtRoSs"
+// var testString2 = "jabberwocky"
+// // Expected output: "jAbBeRwOcKy"
+
+
+
+describe('when everyOther is called', () => {
+    it('you should get a return with every other letter capitalized', () => {
+        let string = "albatross"
+        let newString = everyOther(string)
+        // Expected output: "aLbAtRoSs"
+        expect(newString).toEqual("aLbAtRoSs")
+    })
+    it('you should get a return with every other letter capitalized', () => {
+        let string2 = "jabberwocky"
+        let newString2 = everyOther(string2)
+        // Expected output: "jAbBeRwOcKy"
+        expect(newString2).toEqual("jAbBeRwOcKy")
+    })
+}) 
 // b) Create the function that makes the test pass.
+// first i gotta split the string into an array
+// iterate through the array an dfind the index of each value
+// if the value is odd, .toUpperCase
+// join the array back into a string
+
+const everyOther = (strr) => {
+    let splitStrr = strr.split("");
+    everyOtherValue = splitStrr.map((value, index) => {
+        if (index % 2 != 0) {
+            return value.toUpperCase();
+        } else {
+            return value;
+        }
+    })
+    return everyOtherValue.join('');
+}
+
+
 
 
 
 
 
-// --------------------5) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
-// a) Create a test with an expect statement using the variables provided.
 
-var testArray1 = [3, 7, 10, 5, 4, 3, 3]
-var testArray2 = [7, 8, 2, 3, 1, 5, 4]
-// Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
 
 
 
 
 
+
+
+// // --------------------5) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
+
+// // a) Create a test with an expect statement using the variables provided.
+
+// var testArray1 = [3, 7, 10, 5, 4, 3, 3]
+// var testArray2 = [7, 8, 2, 3, 1, 5, 4]
+// // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
+
+
+test('when noDuplicates is called, you should get a new array with no duplicates', () => {
+    let testArray1 = [3, 7, 10, 5, 4, 3, 3]
+    let testArray2 = [7, 8, 2, 3, 1, 5, 4]
+        // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
+        expect(noDuplicates(testArray1, testArray2)).toEqual([3, 7, 10, 5, 4, 8, 2, 1])
+    })
 // b) Create the function that makes the test pass.
+
+const noDuplicates = (array1, array2) => {
+    noDoubles = [];
```

Comment:
global variables are a bit icky in JS because they can cause conflicts. try declaring variables using var, let, const (depending on its usage)